### PR TITLE
Increased test coverage

### PR DESCRIPTION
**Pull request by klesniak1**

### CODE REVIEW

File: src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java [hunk 1]
```diff
@@ -273,6 +273,50 @@ public void buildsTheSpecUsingMultiValueWebParameterAndParamSeparator() throws E
...
+
+        resolver.buildSpecification(ctx, param.getParameterAnnotation(Spec.class));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionWhenExceptionOccurs(){
+        MethodParameter param = MethodParameter.forExecutable(testMethod("testMethod11"), 0);
+        NativeWebRequest req = mock(NativeWebRequest.class);
+
+        when(req.getParameterValues("theParameter")).thenReturn(new String[] {"val1", "val2,val3,val4", "val5,val6", "val7"});
```
Comment (tkaczmarzyk) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java:312]: why exception is expected in this test? refactor code or at least add an explanation in a comment

File: src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java [hunk 2]
```diff
@@ -273,6 +273,50 @@ public void buildsTheSpecUsingMultiValueWebParameterAndParamSeparator() throws E
...
+
+        assertThat(resolver.supports(param.getParameterAnnotations()[0])).isTrue();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void shouldThrowIllegalStateExceptionWhenDefConfigLengthIsMoreThanOne(){
+        MethodParameter param = MethodParameter.forExecutable(testMethod("testMethod11"), 0);
+        NativeWebRequest req = mock(NativeWebRequest.class);
+
+        when(req.getParameterValues("theParameter")).thenReturn(new String[] {"val1", "val2,val3,val4", "val5,val6", "val7"});
```
[the remark this answers is not in the document]
Comment (klesniak1) [reply src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java:300]: it does, because if no args are specified, specification is going to be null and the method that checks the config length won't be invoked.

File: src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java (hunk 1 above)
Comment (klesniak1) [reply src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java:312]: deleted this test method, i accidently abandoned it without refactoring. the goal of this method was to cover the last catch of this try-catch clause:
```
} catch (NoSuchMethodException e) {
			throw new IllegalStateException("Does the specification class expose at least one of the supported constructors?\n"
					+ "It can be either:\n"
					+ "  3-arg (QueryContext queryCtx, String path, String[] args)\n"
					+ "  4-arg (QueryContext queryCtx, String path, String[] args, Converter converter)\n"
					+ "  5-arg (QueryContext queryCtx, String path, String[] args, Converter converter, String[] config)", e);
		} catch (RuntimeException e) {
			throw e;
		} catch (Exception e) {
			throw new IllegalStateException(e);
		}
```
imo this edge case should not happen and can be invoked only by throwing exception by mock. i think that it doesn't make sense to test it.

File: src/test/java/net/kaczmarzyk/spring/data/jpa/utils/JsonBodyParamsTest.java [hunk 3]
```diff
@@ -154,6 +161,21 @@ public void returnsEmptyValueFromJsonWhenKeyNotPresent() {
...
 	}
 
+	@Test
+	public void returnsEmptyValueWhenRequestBodyIsNull() {
+		//given
+		String json = "{ \"key1\": { \"key2\": \"value\" }}";
+		MockedStatic<JsonParser> jsonParserMockedStatic = mockStatic(JsonParser.class);
+
+		//when
+		jsonParserMockedStatic.when((MockedStatic.Verification) JsonParser.parseString(json)).thenReturn(null);
```
Comment (rdworak37) [inline src/test/java/net/kaczmarzyk/spring/data/jpa/utils/JsonBodyParamsTest.java:171]: is there a case when JsonParser returns plain null? i tested parser for null argument and it throws NPE and for empty string value it returns JsonNull object

File: pom.xml [hunk 4]
```diff
@@ -297,6 +297,11 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-inline</artifactId>
```
Comment (tkaczmarzyk) [inline pom.xml:302]: why do we need it?

File: src/main/java/net/kaczmarzyk/spring/data/jpa/utils/JsonBodyParams.java [hunk 5]
```diff
@@ -43,9 +43,6 @@ public Collection<String> getParamValues(String paramKey) {
 	}
 
 	private static JsonElement getElementByKey(JsonElement rootElement, String key) {
-		if (rootElement == null){
-			return JsonNull.INSTANCE;
-		}
```
Comment (tkaczmarzyk) [inline src/main/java/net/kaczmarzyk/spring/data/jpa/utils/JsonBodyParams.java:48]: why did you removed it from production code when the scope was just additional testing?

File: src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkaczmarzyk) [reply src/test/java/net/kaczmarzyk/spring/data/jpa/web/SimpleSpecificationResolverTest.java:300]: does it have to have so many values? It looks like a value-set prepared for other test scenario and does not introduce value for this test. Can you simplify it?

**Review by tkaczmarzyk — APPROVED** [review]
